File: programmers/level2/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda route: route[1])
    len_routes = len(routes)
    idx = 0
    while idx < len_routes:
        temp = routes[idx][1]
        idx += 1
        while idx < len_routes and routes[idx][0]<=temp<=routes[idx][1]:
            idx += 1
        answer += 1
        
    
    return answer

File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0,idx = 0;
        Arrays.sort(routes,(o1,o2)->{
           return o1[1] - o2[1];
        });
        
        while(idx < routes.length){
            int end = routes[idx++][1];
            while(idx < routes.length && inRange(routes[idx],end)){
                idx++;
            }
            answer++;
        }
        return answer;
    }
    
    public boolean inRange(int[] range, int target){
        return range[0] <= target && target <= range[1]; 
    }
    
}